client/src/app/pages/applications/components/application-detail-drawer/components/assessed-archetypes.tsx

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { Ref } from "@app/api/models";
+import { Label, LabelGroup, Spinner } from "@patternfly/react-core";
+import { EmptyTextMessage } from "@app/components/EmptyTextMessage";
+import { useTranslation } from "react-i18next";
+import { useFetchArchetypes } from "@app/queries/archetypes";
+import { useFetchAllAssessmentsWithArchetypes } from "@app/queries/assessments";
+
+interface IAssessedArchetypesProps {
+  archetypeRefs: Ref[] | undefined;
+}
+
+export const AssessedArchetypes: React.FC<IAssessedArchetypesProps> = ({
+  archetypeRefs,
+}) => {
+  const { t } = useTranslation();
+  const { archetypes, isFetching: isFetchingArchetypes } = useFetchArchetypes();
+  const applicationArchetypes = archetypes.filter(
+    (archetype) => archetypeRefs?.some((ref) => ref.id === archetype.id)
+  );
+
+  const {
+    assessmentsWithArchetypes,
+    isLoading: isFetchingAllAssessmentsWithArchetypesLoading,
+  } = useFetchAllAssessmentsWithArchetypes(applicationArchetypes || []);
+  const filteredArchetypes = assessmentsWithArchetypes
```

Comment:
These are assessed archetypes where at least one of the assessments is required.  I think this matches the PR description, but it was a bit confusing at first look.